marker_api/src/ast/common/span.rs

```diff
@@ -147,3 +158,73 @@ impl From<SpanId> for SpanOwner {
         Self::SpecificSpan(id)
     }
 }
+
+pub struct Ident<'ast> {
+    _lifetime: PhantomData<&'ast ()>,
+    sym: SymbolId,
+    span: SpanId,
+}
+
+impl<'ast> Ident<'ast> {
+    pub fn name(&self) -> &str {
+        with_cx(self, |cx| cx.symbol_str(self.sym))
+    }
+
+    pub fn span(&self) -> &Span<'ast> {
+        with_cx(self, |cx| cx.get_span(self.span))
+    }
+}
+
+#[cfg(feature = "driver-api")]
+impl<'ast> Ident<'ast> {
+    pub fn new(sym: SymbolId, span: SpanId) -> Self {
+        Self {
+            _lifetime: PhantomData,
+            sym,
+            span,
+        }
+    }
+}
+
+impl<'ast> std::fmt::Debug for Ident<'ast> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Ident")
+            .field("name", &self.name())
+            .field("span", &self.span())
+            .finish()
+    }
+}
+
+macro_rules! impl_ident_eq_for {
+    ($ty:ty) => {
+        #[allow(unused_lifetimes)]
+        impl<'a, 'ast> PartialEq<$ty> for Ident<'ast> {
+            fn eq(&self, other: &$ty) -> bool {
+                self.name().eq(other)
+            }
+        }
+    };
+}
+
+use impl_ident_eq_for;
+
+impl_ident_eq_for!(str);
+impl_ident_eq_for!(String);
+impl_ident_eq_for!(std::ffi::OsStr);
+impl_ident_eq_for!(std::ffi::OsString);
+impl_ident_eq_for!(std::borrow::Cow<'a, str>);
+
+// The following implementation sadly causes `error[E0210]`, which means that
+// marker can only provide the implementation `<ident> == <string>` and not the
+// other way around. This is a bit meh when it comes to usability, but IDK how to
+// fix it right now (~xFrednet)
+// ```
+// impl<'ast, T> PartialEq<Ident<'_>> for T
+// where
+//     T: PartialEq<str>,
+// {
+//     fn eq(&self, other: &Ident<'_>) -> bool {
+//         self.eq(other.name())
+//     }
+// }
+// ```
```

Comment:
```suggestion
macro_rules! impl_ident_eq_for {
    ($ty:ty) => {
        impl<'ast> PartialEq<$ty> for Ident<'ast> {
            fn eq(&self, other: &$ty) -> bool {
                self.name().eq(other)
            }
        }

        impl<'ast> PartialEq<Ident<'ast>> for $ty {
            fn eq(&self, other: &Ident<'ast>) -> bool {
                other.name().eq(self)
            }
        }
    };
    ($($ty:ty),+) => {
        $(
            impl_ident_eq_for!($ty);
        )+
    };
}

use impl_ident_eq_for;

impl_ident_eq_for![
    str,
    String,
    std::ffi::OsStr,
    std::ffi::OsString,
    std::borrow::Cow<'_, str>
];
```
I think that should work :)

Reply:
... Why does it work so easily now? I understand how, but still, before I tested it for a few minutes, and it didn't work then -.-